### PR TITLE
NPHardEval Leaderboard

### PR DESCRIPTION
**Pull request by clefourrier**
Light review should be sufficient :)

### CODE REVIEW

**Review by lizhouf — APPROVED** [review]

File: leaderboards-on-the-hub-nphardeval.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    <img src="https://github.com/casmlab/NPHardEval/raw/main/figure/questions_blog.png" alt="Weighted Accuracy and Failure Rate" style="width:80%">
+</div>
+
+There are 2 types of data structure: graph data (e.g., GCP) and linear data (e.g., SAS). The synthesis process in both cases is governed by a progression of complexity across a spectrum of predefined levels. Examples are provided below:
+
+<div align="center">
+    <img src="https://github.com/casmlab/NPHardEval/raw/main/figure/questions_examples.png" alt="Weighted Accuracy and Failure Rate" style="width:80%">
+</div>
+
+More background and insights are available in [Slides](https://docs.google.com/presentation/d/1VYBrCw5BqxuCCwlHeVn_UlhFj6zw04uETJzufw6spA8/edit?usp=sharing).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:49]:
```suggestion
More background and insights are available in [these slides](https://docs.google.com/presentation/d/1VYBrCw5BqxuCCwlHeVn_UlhFj6zw04uETJzufw6spA8/edit?usp=sharing).
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Evaluation Metrics
+
+To evaluate the reasoning ability of LLMs, we utilize two metrics, the Weighted Accuracy and the Failure Rate.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:53]:
```suggestion
We use two metrics to evaluate the reasoning ability of LLMs: Weighted Accuracy and Failure Rate.
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+### Weighted Accuracy (WA)
+
+When evaluating problem-solving accuracy, we use a metric called **Weighted Accuracy (WA)**. This method is applied for each problem, either through comparison with a correct answer or via step-by-step result checking for problems without a singular answer. To reflect comparative accuracy more effectively, we assign weights to different difficulty levels. Each level's weight corresponds to its relative importance or challenge, with higher difficulty levels receiving more weight in a linear progression (for instance, level 1 has weight 1, level 2 has weight 2, and so on).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:57]:
```suggestion
**Weighted Accuracy (WA)** is used to evaluate problem-solving accuracy. This method is applied to each problem, either through comparison with a correct answer or via step-by-step result checking for problems without a singular answer. To reflect comparative accuracy more effectively, we assign weights to different difficulty levels. Each level's weight corresponds to its relative importance or challenge, with higher difficulty levels receiving more weight in a linear progression (for instance, level 1 has weight 1, level 2 has weight 2, and so on).
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+The formula for Weighted Accuracy is as follows:
+
+$$
+WA = \frac{\sum\limits_{i=1}^{10} (w_i \times A_i)}{\sum\limits_{i=1}^{10} w_i}
+$$
+
+In this equation, $w_i$ represents the weight assigned to difficulty level $i$ (ranging from 1 to 10), and $A_i$ is the accuracy at that level.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:65]:
```suggestion
In this equation, \\(w_i\\) represents the weight assigned to difficulty level \\(i\\) (ranging from 1 to 10), and \\(A_i\\) is the accuracy at that level.
```

I _think_ this works for inline latex too.

File: leaderboards-on-the-hub-nphardeval.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+### Weighted Accuracy (WA)
+
+When evaluating problem-solving accuracy, we use a metric called **Weighted Accuracy (WA)**. This method is applied for each problem, either through comparison with a correct answer or via step-by-step result checking for problems without a singular answer. To reflect comparative accuracy more effectively, we assign weights to different difficulty levels. Each level's weight corresponds to its relative importance or challenge, with higher difficulty levels receiving more weight in a linear progression (for instance, level 1 has weight 1, level 2 has weight 2, and so on).
+
+The formula for Weighted Accuracy is as follows:
+
+$$
+WA = \frac{\sum\limits_{i=1}^{10} (w_i \times A_i)}{\sum\limits_{i=1}^{10} w_i}
+$$
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:63]:
```suggestion
\\( WA = \frac{\sum\limits_{i=1}^{10} (w_i \times A_i)}{\sum\limits_{i=1}^{10} w_i} \\)
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+Another critical metric we consider is the **Failure Rate (FR)**. This measure helps assess the frequency of unsuccessful outcomes across different problems and difficulty levels. It's particularly useful for identifying instances where an LLM's result does not match the expected output format.
+
+The Failure Rate is calculated by considering the proportion of failed attempts relative to the total number of attempts for each difficulty level. An attempt is counted as failed if the model generates results that cannot be successfully parsed in all endpoint calls. We set the maximum number of tries as 10. For each problem, the Failure Rate is then aggregated across all difficulty levels, considering the total of 10 attempts at each level.
+
+The formal definition of Failure Rate is:
+
+$$
+FR = \frac{\sum_\limits{i=1}^{10} F_i}{100}
+$$
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:77]:
```suggestion
\\( FR = \frac{\sum_\limits{i=1}^{10} F_i}{100} \\)
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+Here, $F_i$ denotes the number of failed attempts at difficulty level $i$.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:79]:
```suggestion
Here, \\( F_i \\) denotes the number of failed attempts at difficulty level \\( i \\).
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+```
+- For its fewshot experiment, 
+```
+  cd Close/run
+  python run_p_EDP_few.py gpt-4-1106-preview self
+```
+
+We currently support fewshot examples from the same question (self), and may support examples from other questions (other) in the future.
+
+## Join the Conversation
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:124]:
```suggestion
## Join the Conversation

```

File: leaderboards-on-the-hub-nphardeval.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+[The NPHardEval leaderboard](https://huggingface.co/spaces/NPHardEval/NPHardEval-leaderboard)'s [dataset](https://github.com/casmlab/NPHardEval/releases) and [code](https://github.com/casmlab/NPHardEval) are available on Github and HuggingFace for community access and contributions.
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:125]:
```suggestion
[The NPHardEval leaderboard](https://huggingface.co/spaces/NPHardEval/NPHardEval-leaderboard), [dataset](https://huggingface.co/datasets/NPHardEval/NPHardEval-results) and [code](https://github.com/casmlab/NPHardEval) are available on Github and Hugging Face for community access and contributions.
```

File: leaderboards-on-the-hub-nphardeval.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+We'll love to see community contributions and interest on the NPHardEval [GitHub Repository](https://github.com/casmlab/NPHardEval) and [HuggingFace Leaderboard](https://huggingface.co/spaces/hyfrankl/NPHardEval-leaderboard).
```
Comment (pcuenca) [inline leaderboards-on-the-hub-nphardeval.md:127]:
```suggestion
We'll love to see community contributions and interest on the NPHardEval [GitHub Repository](https://github.com/casmlab/NPHardEval) and [Hugging Face Leaderboard](https://huggingface.co/spaces/NPHardEval/NPHardEval-leaderboard).
```

**Review by pcuenca — APPROVED** [review]
Thanks a lot!

File: leaderboards-on-the-hub-nphardeval.md [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+NPHardEval introduces a dynamic, complexity-based framework for assessing Large Language Models' (LLMs) reasoning abilities. It poses 900 algorithmic questions spanning the NP-Hard complexity class and lower, designed to rigorously test LLMs, and is updated on a monthly basis to prevent overfitting!
+
+<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.45.1/gradio.js"> </script>
+<gradio-app theme_mode="light" space="NPHardEval/NPHardEval-leaderboard"></gradio-app>
+
+## A Unique Approach to LLM Evaluation
+
+[NPHardEval](https://arxiv.org/abs/2312.14890) stands apart by employing computational complexity classes, offering a quantifiable and robust measure of LLM reasoning skills. The benchmark's tasks mirror real-world decision-making challenges, enhancing its relevance and applicability. Regular monthly updates of the benchmark data points mitigate the risk of model overfitting, ensuring a reliable evaluation. 
+
+In particular, our major contributions are two-fold:
```
[the remark this answers is not in the document]
Comment (clefourrier) [reply leaderboards-on-the-hub-nphardeval.md:27]: Super good point, I'll do the edits accordingly

**Pull request merged**